extension/authoidcextension/extension.go

```diff
@@ -125,20 +128,26 @@ func (e *oidcExtension) Authenticate(ctx context.Context, headers map[string][]s
 		// to read the claims. It could fail if we were using a custom struct. Instead of
 		// swalling the error, it's better to make this future-proof, in case the underlying
 		// code changes
-		return errFailedToObtainClaimsFromToken
+		return ctx, errFailedToObtainClaimsFromToken
 	}
 
-	_, err = getSubjectFromClaims(claims, e.cfg.UsernameClaim, idToken.Subject)
+	// we could have set this right after obtaining the raw auth string, but we should only change the
+	// context if the auth was successful
+	ctx = auth.NewContextFromRaw(ctx, raw)
+
+	sub, err := getSubjectFromClaims(claims, e.cfg.UsernameClaim, idToken.Subject)
 	if err != nil {
-		return fmt.Errorf("failed to get subject from claims in the token: %w", err)
+		return ctx, fmt.Errorf("failed to get subject from claims in the token: %w", err)
 	}
+	ctx = auth.NewContextFromSubject(ctx, sub)
 
-	_, err = getGroupsFromClaims(claims, e.cfg.GroupsClaim)
+	groups, err := getGroupsFromClaims(claims, e.cfg.GroupsClaim)
 	if err != nil {
-		return fmt.Errorf("failed to get groups from claims in the token: %w", err)
+		return ctx, fmt.Errorf("failed to get groups from claims in the token: %w", err)
 	}
+	ctx = auth.NewContextFromMemberships(ctx, groups)
```

Comment:
Presumably this should be only done if the authenticator is explicitly configured by the user to do so. I do not think we need to do this always. So for example:
```yaml
extensions:
  oidc:
    passthrough: true # if this is true only then we store the auth data in the context
```

Reply:
How about this being a property of the receiver?

```yaml
receivers:
  otlp/auth:
    protocols:
      grpc:
        endpoint: localhost:4318
        auth:
          propagate: true
          authenticator: oidc
```

Note that we will have a `propagatingauthextension` as client auth as part of #2734, which will propagate the auth data to the destination backend, but I think you didn't mean that case.

Reply:
> How about this being a property of the receiver?

Hmm, probably, not sure what it would look like in the code. But, is that more desirable than specifying it in the extension?

Reply:
> Note that we will have a `propagatingauthextension` as client auth as part of #2734, which will propagate the auth data to the destination backend, but I think you didn't mean that case.

I am confused. What is `propagatingauthextension`? I do not see it mentioned anywhere.